### ISSUE TITLE
Critical Rolls and Fumbles

### ISSUE THREAD

**Issue by pedroaugustobt** [AnthonyMonette/TheWitcherTRPG#325]
In an earlier AM build of the system, when a roll was a Critical or a Fumble, you had to click the green / red bar to get it to then roll - this seemed to give the system time to then roll the die using the Dice so Nice / Dice tray, which my group loved happen on the board- especially when the Dice 'exploded' into a BIG crit!

Also, when they'd seen that they fumbled, they had to click to roll the next Dice to determine the fumble total. At this point, most of my group would apply Luck to offset the next scary roll.

The auto rolling of Crits and Fumbles, whilst keeping the game running more quickly, took that option away, so we just house rule it by ignoring the auto-roll player fumbles and allowing them to apply luck and roll a new D10 to determine it, and then accepting that roll.

I'm taking a few look at the codes and I understand that when the die explodes, it throws the printed result directly into the chat, without rolling the exploded die or the previous one.

**Comment by ortegamarcel** [comment]
I already tried it once to fix that. The problem is the way the roll logic is implemented, so that it is not really compatible with Dice so Nice. Instead it would be necessary to completely rewrite the function. That would be possible, but it could lead to some unexpected behaviour on other places, that's why I for now only included it in the [Token Action HUD TheWitcherTRPG - Module](https://foundryvtt.com/packages/fvtt-token-action-hud-thewitchertrpg). There you have the option in the settings to use the "new" dice-rolling-logic. That will also show a animation on crits/fumbles. But they are fully automated.

**Comment by pedroaugustobt** [comment]
> I already tried it once to fix that. The problem is the way the roll logic is implemented, so that it is not really compatible with Dice so Nice. Instead it would be necessary to completely rewrite the function. That would be possible, but it could lead to some unexpected behaviour on other places, that's why I for now only included it in the [Token Action HUD TheWitcherTRPG - Module](https://foundryvtt.com/packages/fvtt-token-action-hud-thewitchertrpg). There you have the option in the settings to use the "new" dice-rolling-logic. That will also show a animation on crits/fumbles. But they are fully automated.

How do i configure this new dice-rolling-logic? Not working for me...

I downloaded your latest version

**Comment by ortegamarcel** [comment]
![foundry_token-action-hud-setting](https://github.com/AnthonyMonette/TheWitcherTRPG/assets/9929801/a5d9514e-a5c2-4bf9-82ee-8e3039309f9a)

Go to settings and uncheck the box on "Roll Skills Natively". This should be the default if you are using TheWitcherTRPG v0.96.

**Comment by pedroaugustobt** [comment]
![image](https://github.com/AnthonyMonette/TheWitcherTRPG/assets/107325727/be43b7ea-daef-4b44-b4a7-08534ee09423)
Does this work in 0.97? I unchecked the option and it's not working, it doesn't roll the dice and prints in the chat instantly, I couldn't find any error log in the console.

**Comment by ortegamarcel** [comment]
Looks like 0.97 is not compatible with Foundry v10. Either downgrade the systems version to 0.96 or update Foundry to v11.

BTW: When you have more problems with Token Action HUD please report them [here](https://github.com/ortegamarcel/fvtt-token-action-hud-thewitchertrpg/issues).

**Comment by AnthonyMonette** [comment]
One thing that could be possible is to add a flag in the system setting to automatically roll crit/fumble  and when this option is checked-out it goes back to the previous behavior where user would need to click on the button to roll the crit/fumble.  What do you think?

**Comment by pedroaugustobt** [comment]
> One thing that could be possible is to add a flag in the system setting to automatically roll crit/fumble  and when this option is checked-out it goes back to the previous behavior where user would need to click on the button to roll the crit/fumble.  What do you think?

Yeah, its a solution, but when you craft an item with a high DC and player rolls 10 and still doesn't pass, the item will not be crafted.

Currently i deleted a line of code where it calculates the previous value taken on the dice + the current value of the explosion, this briefly resolved It, but the dice that appears in the chatMessage os the total of the firts dice rolled.

**Comment by pedroaugustobt** [comment]
The error is in the part where it prints the summed result, it behaves by rolling the dice under the table, thus cutting the Dice so Nice animation and printing the result in the chatMessage.
